src/parallel/job_token.rs

```diff
@@ -233,19 +233,14 @@ mod inprocess_jobserver {
     impl JobServer {
         pub(super) fn new() -> Self {
             // Use `NUM_JOBS` if set (it's configured by Cargo) and otherwise
-            // just fall back to a semi-reasonable number.
-            //
-            // Note that we could use `num_cpus` here but it's an extra
-            // dependency that will almost never be used, so
-            // it's generally not too worth it.
-            let mut parallelism = 4;
-            // TODO: Use std::thread::available_parallelism as an upper bound
-            // when MSRV is bumped.
-            if let Ok(amt) = var("NUM_JOBS") {
-                if let Ok(amt) = amt.parse() {
-                    parallelism = amt;
-                }
-            }
+            // just fall back to the number of cores on the local machine.
+
+            let num_jobs = var("NUM_JOBS").ok().and_then(|j| j.parse::<u32>().ok());
+            let parallelism = num_jobs.unwrap_or_else(|| {
+                std::thread::available_parallelism()
+                    .map(|v| usize::from(v) as u32)
+                    .unwrap_or(4)
+            });
 
             Self(AtomicU32::new(parallelism))
```

Comment:
```suggestion
            let parallelism = var("NUM_JOBS")
                .ok()
                .and_then(|j| j.parse::<u32>().ok())
                .or_else(|| Some(std::thread::available_parallelism().ok()?.get() as u32))
                .unwrap_or(4);
```

@madsmtm wdyt

Reply:
I think that the amount of useful bikesheddin on a PR of this size has been spent xD This looks fine, I'll apply that.